cmd/argocd-applicationset-controller/commands/applicationset_controller.go

```diff
@@ -60,24 +60,28 @@ func NewCommand() *cobra.Command {
 		Use:   "controller",
 		Short: "Starts Argo CD ApplicationSet controller",
 		RunE: func(c *cobra.Command, args []string) error {
+			vers := common.GetVersion()
+			namespace, _, err := clientConfig.Namespace()
+			errors.CheckError(err)
+			log.WithFields(log.Fields{
+				"version":   vers.Version,
+				"commit":    vers.GitCommit,
+				"built":     vers.BuildDate,
+				"namespace": namespace,
+			}).Info("ArgoCD ApplicationSet Controller is starting")
+
```

Comment:
Can/should this be pulled into a util function?

Reply:
Originally, I did that. I originally made `common.Version` a receiver for a new `LogStartupInfo` function, but then as it turns out, `namespace` isn't available or relevant information for all components. Similarly, port number isn't relevant for all components. I found myself putting conditions like `if namespace == ""` and  `if port == 0` in the function and then realized that `namespace` or `port` have nothing to do with a `common.Version`, so I went back to doing things like this...

But now that you're asking about this, you're making me reconsider. I should try something like the following instead:

```golang
func (v Version) LogStartupInfo(componentName string, fields log.Fields) {
	fields["version"] = v.Version
	fields["commit"] = v.GitCommit
	fields["built"] = v.BuildDate
	log.WithFields(fields).Infof("%s is starting", componentName)
}
```

I'll play with this a bit and amend the PR.

Reply:
I've amended the PR as discussed above.